### DISCUSSION TITLE
DNF and its relationship with member projects

### DISCUSSION THREAD

**Discussion by Aaronontheweb** [dotnet-foundation/Home#38] [Members Talk]
It's been tough to miss the disturbing reports from some of our member projects that the DNF has, in essence, abused its administrative access to these projects:

- https://www.glennwatson.net/posts/dnf-problems-solutions
- https://github.com/reactiveui/splat/pull/778 
- https://twitter.com/JeremySkinner/status/1445272800185495555

And there's Rodney's recent post clarifying the... frankly, dishonest way his departure from the foundation was explained via the official mailing list: https://rodneylittlesii.com/posts/topic/foundation-echo-chamber - among other issues he highlights.

The more I hear, the more I'm outraged at the foundation's behavior. The set of expectations that were made to our project was that we were to be left to our own devices, but the behavior of the foundation leadership has become increasingly erratic, uncommunicative, and fait accompli.

Transferring projects out of public Github to the .NET Foundation's Enterprise Github account without so much as notifying them? Blocking their signing keys on NuGet? Forcing pull requests over the objections of maintainers? 

I'm far from alone among DNF maintainers in saying that the behavior that's been exposed here has deeply undermined my confidence and trust in the foundation going forward. 

What's the justification for any of this? Why have these steps been taken without any communication with the maintainers who oversee day-to-day operations of these projects?

**Comment by robmen** [comment]
August 3rd, I got this email:

> Yesterday we announced Foundation-wide Code of Conduct Enforcement. Part of making that work requires that the dnfadmin GitHub user has owner permissions to GitHub organizations. GitHub org ownership by the dnfadmin account has been required for all Foundation projects for some time, but we noticed that the account does not have the proper permissions in wixtoolset. Please assign the org ownership role to the dnfadmin user as soon as possible. 

I was busy on a major deadline and responded on September 14th:

> Thanks for reaching out. The bulk of our day to day conversations happen on a couple mailing list (wix-users@lists.wixtoolset.org and wix-devs@lists.wixtoolset.org) that are (fortunately) very civil and quite installation-geeky. But we could/would unsubscribe anyone that violated the Code of Conduct there.

> Also, the WiX Toolset’s repositories are too important to us to grant ownership access to an account that isn’t a core maintainer of the codebase. So, thank you for the offer to automate the Code of Conduct enforcement but we’ll pass if it requires significant access to repo permissions.

I think I made it explicitly clear that I did not trust the .NET Foundation with admin access to the repository.

However, another of our maintainers was blocked from approving a PR due to the clabot being broken. He was told the fix was to make @dnfadmin an owner of the project.

Apparently, I made a fatal mistake on September 21st by granting the @dnfadmin access to the @wixtoolset organization to get the CLA bot fixed.

I just read Glenn's blog post gobsmacked and thought to check the @wixtoolset organization. I was floored. We too were moved into the .NET Foundation's GItHub Enterprise.

We were moved into the .NET Foundation's GitHub Enterprise without discussion *after* I explicitly called out I did not trust the organization.

I am so angry. I feel so betrayed.

**Reply by FransBouma** [reply]
@robmen The (c) of the WiX source is with the DNF according to the License.txt file in the repo. This is what I find strange: why did the WiX org sign over the copyright to the DNF, while e.g. you don't trust them enough to have control over the repo? *if* they own the code, it kinda makes sense to have control over it too (as they own it, through the copyright), but reading your post it looks like this copyright transfer might have happened unintentional (or have happened at all... changing the (c) in a License.txt file doesn't sign over the copyright for e.g. the copyright law in many EU states, it has to be done by signed contract/notary act)

FTR: I agree with you that this transfer of the org without any discussion is totally wrong.

**Reply by JeremySkinner** [reply]
@FransBouma projects that used the "Contribution" model when joining the foundation didn't transfer copyright to the foundation (they license the project to the foundation instead), although the foundation still likes to have "(c) .net foundation and contributors" in the license headers (it's the same for my project), although whether this is actually necessary I'm not sure (maybe it's not? I'm not an expert), so even if it's in the project header that doesn't necessarily mean the foundation owns the copyright, but rather which agreement the owner signed when joining the foundation (contribution vs assignment). Not sure if that's the case for wix.

**Reply by FransBouma** [reply]
@JeremySkinner ok, so when the project didn't transfer the (c) officially to the foundation, there shouldn't be a (c) .NET foundation in the license file, as they don't own it; If they don't own it, they can't do anything to the code either, be it moving it to another repo, change the license etc. But if the copyright is transferred officially to the DNF, then they can do whatever they want with it (and they likely will, as from their PoV it's of course not good if they own code (and thus are responsible for) but can't do anything with it). 

It might be important to your (and other projects) to get official word (from e.g. a notary/lawyer) who owns the copyright of the code you're maintaining. If you own the copyright, I'd change the (c) header everywhere to your name and not theirs. After all, you own it, not the DNF. 

From robmen's post I can only conclude he definitely didn't willingly sign over the (c) of the sourcecode.

**Reply by gortok** [reply]
The .NET Foundation has Microsoft’s lawyers on its side. If you’re going to sign a contract you need a lawyer too. Whether or not you assigned the copyright to them, it’s effectively assigned to them when you change the copyright to their name and “contributors”. You’d have a hard time asserting a copyright legally without your name there. Maybe this was inadvertent, but it’s not in your best interest to change these things without talking to a lawyer that represents your interests.

**Reply by robmen** [reply]
The WiX Toolset was Microsoft's first OSS project. I never owned the copyright to the code because I was employed by Microsoft. The copyright passed through a couple of organizations to eventually land at .NET Foundation.

But I (and other maintainers of the WiX Toolset) do all of the work in our GitHub organization `wixtoolset`. The .NET Foundation can maintain a fork of the source code, or archive it, or print it out and admire it however they want. But I don't want people I don't trust in position to delete all of the hard work that we do in our `wixtoolset` organization.

**Reply by FransBouma** [reply]
@robmen ah thanks for the clarification. The sad thing is tho, that as they own the copyright, then it seems reasonable they get access to the source code so they can do with it what they need to (as they legally own it). However it's obviously reasonable to discuss things with the people who actually do the work, fully agreed with that. How they handled this is beyond sad and doesn't bode well for the future (I mean, who in their right mind would trust these people with their code now? I wouldn't (I never have, but that's another story)). 

As they own the copyright, in theory they can do as they please with the code; delete things, add stuff themselves, the works, that is to the code they own, not the code you contributed when outside Microsoft (unless that's in the contract WiX has with DNF). 

All in all a sad affair and if the DNF doesn't change course with how they handle the backbone of their right of existence (the maintainers of the project they govern) I don't see how this can end well. Good luck, you have my support (how little that might mean in the grand scheme of things).

**Reply by poke** [reply]
I'm not sure what owning the code (whatever exactly that is supposed to mean with open source licenses) has to do with them gaining (admin) access to a GitHub organization or even the project itself. If I fork a DNF project, they cannot administer my fork either, even though me forking it changed nothing about copyrights, ownership or licenses.

**Reply by IanKemp** [reply]
> However, another of our maintainers was blocked from approving a PR due to the clabot being broken. He was told the fix was to make @dnfadmin an owner of the project.

> Apparently, I made a fatal mistake on September 21st by granting the @dnfadmin access to the @wixtoolset organization to get the CLA bot fixed.

I honestly don't understand why you'd ever even do this... there's no logical explanation for why a CLA bot would require a completely different user to have any special access to your project, **especially** ownership rights! If the answer to "how do I fix feature X" is "give user Y god mode", your alarm bells should be going off regardless of the trust relationship between you and Y.

**Reply by FransBouma** [reply]
@poke owning the code is owning the copyright of the code, as stated in the copyright law that's applicable on the location the code is used in. OSS licenses are about distribution and usage, not about copyright. You never transfer copyright through an OSS license, you merely give rights to use the code and distribute it in a certain shape/form. 

If you own the copyright of a codebase, you can do whatever you want with it, also revoke OSS (so distribution) licenses, move it elsewhere etc. If you fork a repo, you can alter the code, but you don't own it, you might think you can change the license, but that's not true; only the copyright holder can. That's why it's critical to know who owns a codebase in this case, if the DNF moves a codebase to a different location or does other things with it that are allowed only by the copyright holder.

**Reply by matkoniecz** [reply]
See https://robmensching.com/blog/posts/2021/10/6/how-the-.net-foundation-kerfuffle-became-a-brouhaha/ for how this story ended. Noteworthy parts:

> If you are not part of the GitHub Enterprise, you can't tell if an org is part of "public" GitHub or a GitHub Enterprise.

so regular user cannot check whether projects stolen and silently transferred as part of this supply chain attack are back with real maintainers. And answer to "So, how did you leave the .NET Foundation's GitHub Enterprise?" turned out to be a loophole in settings.

Congratulations on recovering your repositories!

See also https://github.com/dotnet-foundation/Home/discussions/59 where they started allowing to recover repositories, but only after an explicit request. And after blog post mentioned above.

**Reply by glennawatson** [reply]
They have been transferring repositories to the public instance. They been doing opt out for greater transparency. Eg there is potential billing implications etc if you have private repos or LFS data.

**Comment by leastprivilege** [comment]
After reading these tweets and articles I had a look at our organization page. To my surprise, I found this new link at the top:

<img width="1084" alt="Screenshot 2021-10-06 at 08 37 00" src="https://user-images.githubusercontent.com/1454075/136197208-ce3329ba-4943-4433-9b72-8d0b0af2d039.png">

When I follow this link, it brings me to https://github.com/enterprises/dotnet

My questions are:

* What is this and what does it mean?
* Why was this change made without our prior consent or discussion?
* How can do we undo this change?

thanks

**Reply by brianlagunas** [reply]
Essentially it means that ownership of your org has been transferred to DNF without you ever knowing. Mainly because the org/repo links stay the same. There is no other way to know this happened unless you go looking for it specifically.

The same thing happen to the Prism Library org. I had no idea this ever happened. This was never communicated to me before or after it happened.

**Reply by leastprivilege** [reply]
Interesting. How is it even possible to transfer ownership of an org without consent of the owner?

**Reply by brianlagunas** [reply]
They did it using the dnfadmin account which was given the owner role to your org

**Reply by FransBouma** [reply]
The root question is *why*, though. That they used that account doesn't mean they should, the account is also necessary for other housekeeping duties, right? 
To me this particular case is a cornerstone to distrust the DNF in every action they do, and especially the lack of *any* communication with the people who actually form the project: the maintainers.

**Reply by nathan-alden-sr** [reply]
A theory:
1. They considered "if we tell maintainers we're going to do this, they will resist and likely take immediate action to prevent it, or raise a stink."
2. They feel convicted that they own the code and the management of said code.
3. They decided the only course of action was to secretly, and with automation, transfer ownership that likely can't be reverted by maintainers.

It makes sense if you think about it.

**Reply by IanKemp** [reply]
I personally don't see any evil intent here - just standard Microsoft stupidity, incompetence, and failure to communicate. When it comes to large organisations, Hanlon's razor almost always applies.

**Reply by TechnikEmpire** [reply]
Embrace. Extend. Extinguish. Frankly though in this case, I'm happy to see it since I can't use blazor without having a multi-thousand-dollar-per-year, non-oss layer deeply integrated into the microsoft libraries.

**Comment by MeikTranel** [comment]
Just a reminder: we're still talking in a public forum. I was pointed here after asking why the internal member discussions were disabled by the admins and if there is a new place for those discussions - it certainly is not internal.
Every comment here is visible to the public and might just amplify certain opinions that aren't necessarily a good representation of the dnf membership; maybe even just push some people to express themselves differently that they would without public scrutiny. (may i remind everyone here that part of this drama started because people were brigrading from twitter etc.)

**Reply by mgravell** [reply]
you are right, of course, although a large part of the problem here in the first place was a *lack* of transparent communication; but: I have deleted my comment - I think the point is already made, and it doesn't need me pouring kerosene on it; I wanted to say that here, just in case anyone thought that "they" had deleted my comment to silence me; I'll go back to quietly waiting for an official response

**Reply by Aaronontheweb** [reply]
I started the thread precisely because it's public. And this isn't "drama" - my OSS project, part of the .NET Foundation, puts food on the table for myself and my employees. I have real skin in the game here - as do many of the other maintainers chiming in on this thread. 

This isn't an issue because people were pointlessly whining on Twitter - it's because evidence pointing to a pattern of miscommunication, abuse of trust, and dishonesty (see Rodney's post) was exposed there. The opinions you're worried about being "amplified" are from the people whose livelihoods are directly impacted by this behavior.

**Reply by schneidenbach** [reply]
Two things:

1. Echoing what Aaron said - this isn't "drama", and qualifying the community's response to the Foundation as such minimizes the very real issues with the Foundation's handling of its member projects.
2. Part of the issue with the Foundation's operations is the lack of transparency. If anything, having this discussion in a public forum is an improvement.

**Reply by gortok** [reply]
The elephant here is that the private and internal forums were shut down. This is what we have.  There's been no word from the board since this stuff started dropping last week (with the ReactiveUI self-merge by the ED), and there have been *plenty of opportunities* for either 1) members of the board or 2) for the executive team to say something.

No one has said a word. That's not good.

**Reply by MeikTranel** [reply]
Maybe calling it "drama" is insensitive to those with "skin in the game" - i can definitely see that and i want to apologize if that made the relevant people feel less than or something.
But i vehemently disagree that transparency will be achieved by dragging this out in the public. Most of the DNF membership (being silent members who vote and observe discussions here and there) first heard of this by way of a twitter thread starting with @RLittlesII post. We've had people who threatened to resign their membership in comments over this before the majority of members even had the chance to internally demand a statement let alone talk about it (Part of this seems to be related to the fact that we currently simply have no accessible forum to even do that - that's why my first reaction was why is the old members discussion channel disabled by admins).
I wholeheartedly agree that most of what occurred needs to be addressed... fast, but i really don't think it is very helpful to discuss this when there's a constant stream of bad faith discussions happening in separate threads. You will not get anything from the current leadership except empty excuses if the board has to worry about marketing above everything else. You will not get opinions and stances only damage reduction.

@gortok couldn't agree more.

**Reply by glennawatson** [reply]
@meiktranel mostly agree with your sentiment. More focus on constructive feedback is definitely needed. Both myself and Rodney set out to get some answers from the board (we work together a lot) and we deliberately framed it more here are the facts and here's what we'd like answered.

**Reply by jnm2** [reply]
> You will not get anything from the current leadership except empty excuses if the board has to worry about marketing above everything else. You will not get opinions and stances only damage reduction.

Maybe negative publicity makes a mature response from the foundation less likely, but I see no reason why it rules it out. The burden is in a predictable and natural place at the moment. I would hope for the foundation to rise above it even if there are bad-faith discussions provoked by the foundation's actions (and I'm not entirely sure I've seen any bad-faith discussions). The foundation always has the choice to own up to its mistakes rather than doubling down.

**Reply by Aaronontheweb** [reply]
> but i really don't think it is very helpful to discuss this when there's a constant stream of bad faith discussions happening in separate threads. 

The accusations of impropriety, while unpleasant, aren't bad faith because they're substantiated by evidence. The DNF set one expectation when we joined and failed to meet it according to multiple maintainers chiming in on the thread here. The loss of trust is real and substantial - expressing it out loud, even if it's unpleasant to hear or read, doesn't make it bad faith. What we've demanded, thus far is an explanation - which I think you agree is justified.

> You will not get anything from the current leadership except empty excuses if the board has to worry about marketing above everything else. You will not get opinions and stances only damage reduction.

If that's true then it's an abject failure of foundation leadership on two fronts:

* They should have started turning over rocks when the original ReactiveUI PR bruhaha happened a couple of weeks ago, where the Executive Director of the foundation was directly accused of a Code of Conduct violation - instead, the community took the lead and discovered a lot of rot in the process. Hence why this is being done in public.
* If there isn't any public accountability of that failure _and_ the others that lead us here, then the foundation lacks integrity: its word means nothing.

Bill has already weighed in and said the DNF will address this. We shall see.

**Reply by phillip-haydon** [reply]
It bothers me that `the board is meeting` and everything is happening behind closed doors. It's become damage control now rather than reverse what's being done and then openingly discuss what happened. 

Since it's all happening behind closed doors I don't think I can trust DNF at all, period.

**Reply by egil** [reply]
I think its a great thing to have this discussion in public, that's how trust is reestablished in the end. As long as that discussion is based in facts and people are not being personally attacked, but arguments are, then we're just having a normal discourse that is needed to sort out problems.

**Reply by jnm2** [reply]
@phillip-haydon I don't know how a potentially difficult internal discussion could happen if it was forced to be public. There are so many more hangups that get in the way in public and no one feels as safe to say what they think. I've learned on a personal level that private conversations lead to better outcomes.

**Reply by phillip-haydon** [reply]
@jnm2 It happening behind closed doors leads me to believe it's damage control and not wanting to change anything. 

The first thing that should be done is reversing what has being done, that does not need a discussion behind closed doors. Once the reversal has been done that gives a level of trust back to all the maintainers here that something was done accidently without being discussed with maintainers of the projects in the DNF.

If that happens, the board meeting to discuss privately what went wrong, and why, and such, is much easier to accept and move forward.

Effectively in regards to the pulling all the projects under the DNF org is the org saying "all your stuff belongs to us now", and delaying rectification just makes the org more distrustworthly. 

If you pull down a production database by mistake, do you have a meeting to discuss what happened or do you get the database online first?

**Reply by jnm2** [reply]
I don't see it the same way. There should be an internal discussion with time for people to understand and get behind a fuller perspective, rather than knee-jerk actions that relieve the pressure. *That* would be damage control and it's not how I would want the foundation to make decisions.

**Reply by jen20** [reply]
An important reason for this discussion to be held in public, here, rather than on Twitter, is that it grants permanent visibility for those who may consider allowing their work to fall under the .NET Foundation umbrella in future, or for those who may be evaluating whether or not it is acceptable to sign a CLA which assigns copyright.

(Edited to fix minor typo)

**Reply by jnm2** [reply]
Just to clarify, this public discussion is good and I am relieved that it's happening. But the internal discussion (that Philip was referring to) within the foundation before responding back to us is also good.

**Reply by tracker1** [reply]
For what it's worth... Collectivist bargaining isn't a bad think. From boycotts to unionization.  Even if it's brigading, it's not inherently bad.

It just needs to be considered before over reacting wether it's brigading or cancel culture in general.

**Comment by BillWagner** [comment]
I'll provide a brief status from the board:

- The board met yesterday to learn more about what happened, what was and wasn’t communicated, and start working on what we must do better. This was the incoming board, so some members are just getting background.
- The board will issue an apology statement later today. It's being draft now. This will address both the Rx PR, and the GitHubEnterprise account.
- The board is meeting every day this week to make an initial plan to improve communications / discussions with all project maintainers. We obviously need to do better. I'm using "initial plan" here because we want to take some thoughts to all project maintainers. We want to get their thoughts on how and how often to communicate. Once we agree on a plan with the project maintainers, we'll implement it.

**Reply by Aaronontheweb** [reply]
And what about the way Rodney's departure from the DNF was handled?

**Reply by RLittlesII** [reply]
That isn't what this thread is about.  This thread is about the maintainers, and what we need to from the Foundation to be successful.

**Reply by gortok** [reply]
@RLittlesII The thread is about trust, and the handling of your departure plays into that.  I realize that is uncomfortable and it feels like a spotlight, and I can't help that, but that's where we are.

**Reply by Aaronontheweb** [reply]
It's about the integrity of the foundation - if they're willing to misrepresent things or be casually dishonest in their public-facing communications, then that impacts my relationship and trust with them as a maintainer. As much as I like you personally, I want them to take accountability for what happened there because it's really part of the larger issue: our relationship with the foundation.

**Reply by glennawatson** [reply]
Let Rodney drive the narrative on regards to his departure since he knows it best.

**Reply by glennawatson** [reply]
@BillWagner if you could also address the nuget signing issue as well that would be great.

**Reply by Aaronontheweb** [reply]
@glennawatson need to hear the DNF's side of the story.

**Reply by hhariri** [reply]
@BillWagner if you could also address the issue @leastprivilege has brought up, that would be great also.

**Reply by glennawatson** [reply]
That's part of the GitHub enterprise @hhariri

**Reply by hhariri** [reply]
Right, but from my understanding, nobody ever asked permission from the project owners to make this transition.

**Reply by glennawatson** [reply]
Yup but bill already said he's addressing it above

**Reply by JimBobSquarePants** [reply]
> The board will issue an apology statement later today.

It’s been 2 days. Where is the apology?

**Reply by gortok** [reply]
> It’s been 2 days. Where is the apology?

@JimBobSquarePants In case you aren't being snarky, it's here: https://github.com/dotnet-foundation/Home/discussions/39

IF you are being snarky (and I don't blame you if you are), yes, the apology lacked all six parts an apology should have, but given the structure of the Foundation and its relationship with Microsoft; this would appear to be the best they can 'legally' do.

**Reply by JamesRandall** [reply]
If it’s the best they can legally do with regard to an apology imagine how that relationship influences everything else…. If it’s the best they can do with regard to an apology…. good lord.

Edit: for what it’s worth I do think it’s been fed through an organisational acceptability filter. But it also mixed up a personal apology with an organisational one and that’s not ideal because it leaves us all to try and put the line in the right place and land criticism in the right place.

**Comment by uabarahona** [comment]


**Reply by isaacabraham** [reply]
You've got this backwards. Protect themselves? The foundation isn't supposed to be something that needs protecting from. It's supposed to be there to help and support projects, for goodness' sake.

**Reply by jnm2** [reply]
@huunab On the contrary, this being public is an opportunity for the .NET Foundation to show what kind of foundation it's capable of being when confronted with its own miscommunications or mistakes in the future. If it handles it poorly, it's a good thing for everyone that this is public. If it handles it skillfully, it's also a good thing for everyone that this is public.

**Reply by glennawatson** [reply]
Sure having a discussion with stakeholders without all the heat involved would be a smart one. However, it can't always be hiding behind the iron mask, and the outcome must be made public.

**Reply by poke** [reply]
> I would really hate to loose another great initiative like "Identity Server done by the .NET team" or the "Project Maturity Model"

Are you saving that MS building their own identity provider library, directly replacing the IdentityServer from the community would have been a good idea? Are you saying that the rules behind the Project Maturity Model would have been good for the health of the .NET open source ecosystem?

**Reply by poke** [reply]
> probably will never want to provide one, that is […] a hole for the dotnet users.

It is only a hole if you believe that Microsoft (not the foundation, but Microsoft) needs to provide a default implementation for every major use case. This even contradicts the idea behind the maturity model because it would basically prevent community projects from getting to a similar acceptance stage, while also allowing Microsoft to short-circuit the model completely just because _they_ built something. – But I am not going to continue this discussion here again since I believe enough opinions have been collected in that issue about it and Microsoft made their stance clear on the topic.

**Reply by tracker1** [reply]
Personally, I found IdentityServer interesting but overly complicated,. Not using it and creating my own using RSA signed jwt was easier.  In the end, I don't think MS needed to create one.

Azure AD does that from an MS perspective.

That aside, I don't think the maturity model was appropriate given the relationship with maintainers and contributors.  If MS was paying them, maybe... As an OSS community, hell no.

**Comment by CumpsD** [comment]
As an outsider I'm left wondering why you'd keep your project part of the DNF with such an overreach? What do you gain over being a normal OSS project?

**Reply by glennawatson** [reply]
They didn't start getting aggressive with the IP ownership until recently which doesn't line up with the signed contracts or intent. A lot of policies were retroactively applied.

**Reply by CumpsD** [reply]
@glennawatson can you point me, or tell me in short, what a project gains from being under the DNF? 

I realize the current projects are under signed contacts wrt IP, but I find this public discussion very interesting to give insight for other projects which someday might end up under DNF, and give them another view on the pro/cons. (With this thread showing mostly the cons)

**Reply by glennawatson** [reply]
Originally it was resources from the foundation, such as CI pipelines, sometimes Visual Studio Enterprise licenses (no longer granted), code signing for your NuGet packages, some projects found the boost in public presence helpful, legal protection but the code base is MIT anyway.

**Reply by CumpsD** [reply]
That sounds very little for the level of control you're giving away?

**Reply by glennawatson** [reply]
Yeah, at the time of signing up I think a lot of the member projects had a very different relationship with the executive director of the time, and also the view on the legal contracts was very different.

**Reply by rprouse** [reply]
Why join the DNF? As the maintainer of a project that has passed hands several times and has at times seemed dead, I wanted to ensure that someone could pick up after me and continue to develop NUnit if I walked away or if something happened to me. To do that, I knew that I needed to hand over a copy of the keys to everything so it could continue without me.

I ran for the board because I didn't feel that member projects were getting the support they needed and that policies were not being well communicated. Serious mistakes have been made but I truly believe that all current and past members of the foundation and the board have the best intentions. I have known and worked with these people for years and I know their deep commitment to OSS and to the .NET ecosystem.

As a board member I have been silent up until now because I was just elected to the role. When this kicked off I hadn't even attended the board handover. Please forgive this silence but I wasn't willing to speak up until I had a chance to speak to everyone and get my feet under me. Now that I have, I am confident that we can make this right. To do so, we will need to regain the trust of the community and that will be my core focus as a board member. 

To start, we are going to form an OSS maintainers advisory committee. I would like to see all of the foundation's polices concerning member projects vetted by or ideally led by project leads. Once we formalize plans we'll reach out to the community, but if you would like to help, please get in touch.

Several people are saying that we cannot salvage the Foundation because of its DNA but I disagree. The Foundation belongs to us its members and it is up to us to decide what it should be.

**Reply by Aaronontheweb** [reply]
> Please forgive this silence but I wasn't willing to speak up until I had a chance to speak to everyone and get my feet under me

Definitely not an easy first week to begin your tenure, Rob.

Your suggestion of getting the projects more involved in their own governance is a very positive development. I look forward to hearing more about it.

**Reply by glennawatson** [reply]
I've discussed offline with you @rprouse 

* I think fundamentally some maintainers are worried that MSFT is kinda muscling in and attempting to take control. Given there has been so much confusion, there should be mechanisms for projects to opt-in/out. Make something more fitting to what the community was expecting.
* There's a lot of significant changes without any communication about what or why those changes are happening. Allow maintainers to buy into new procedures and policies. 
* Revert the GitHub changes and the NuGet changes. If you need these for the MSFT managed projects apply only to these projects.
* I would stop forcing DNF admin users on maintainers, offer this to maintainers who want the ability to have coverage in case of accident/death etc. More mature projects already have their own redundancy in place.

We were told that there would be more communication after the last major issue with Jon Galloway, so you'd have to follow through with this.

Cheers

**Reply by dggmez** [reply]
@rprouse 

>Several people are saying that we cannot salvage the Foundation because of its DNA but I disagree. The Foundation belongs to us its members and it is up to us to decide what it should be.

I get where you're coming from and I would like to think that's the case. However, I agree with what @dustinmoris said here:

https://github.com/dotnet-foundation/Home/discussions/39#discussioncomment-1442187

After reading the bylaws (that you can argue are a pretty important part of the foundation's DNA) I think that as long as the *Founding Member* has special powers it's pretty hard to claim that the DNF really belongs to the members and not to the *Founding Member*.

**Reply by JamesRandall** [reply]
Completely concur with @dggmez - any changes and steps forward have to be presented within the reality and constraints of what the foundation is. And as @dustinmorris showed it is controlled by its Founding Member and not the members.

If people want it to be controlled by its members then their needs to be a proposal to change the bylaws and remove the special powers belonging to the Founding Member. It would be fascinating to see how that played out. Call me a cynic (realist) but I’m 99% certain that a) it would be blocked and b) if it looked even remotely like happening Microsoft would suddenly become as concerned about copyright assignment and the topics in this thread as regular members.

I’d never paid much heed to the strapline on the foundations homepage until yesterday but looking now it strikes me as disingenuous at best.

**Reply by phillip-haydon** [reply]
> Several people are saying that we cannot salvage the Foundation because of its DNA but I disagree.

I know it may sound like that I think the DNF should disband, but I don't believe that. When Nancy join the DNF I thought even if Nancy didn't need the help/support that being part of the DNF would help promote the idea that the DNF is here to help promote, contribute, support, and do everything within its power with possible financial backing of MS to foster good growth in the .NET OSS community.

Even tho I rarely contribute to stuff these days as real life is too chaotic, I still really want to see the .NET OSS community grow. 

I saw a tweet and I cannot remember who tweeted it but it said that we (the community) should support projects directly. If being part of the DNF is that there's a member fee then some that money should go to help projects via sponsership, especially considering the events that have happened I think alot of projects will absolutely not want to consider joining the DNF in the short term and what we really need is to help these smaller projects that people depend on. 

Right now when the budget: https://dotnetfoundation.org/about/budget

I cannot understand the expenses at all. Half the subscription stuff is going where??? But there's 80k that's sitting somewhere that could be helping projects. 

I obviously have absolutely NO idea how to run a foundation, or what a director and board members do, or a committe. And I'm DEFINITLY not the sort of person who should be representing the DNF. But I do know that I want to help the .NET OSS community and I really wanna see the DNF help! I just don't see it happening at the moment, or in the past :(

**Comment by bjablonsky** [comment]
I'd like to chime in. 

Full disclosure: I'm a member of the .NET Foundation. I am **NOT** an elected member of the board nor have I run for the board. I'm the chairperson of the .NET Foundation Education Committee as I pitched an educational project to the board as a member and was offered the education committee chairperson role since it was vacant. As a chairperson, I have no responsibilities or powers that an elected board member does. These words/thoughts/opinions are my own.

This news is worrisome to hear that open source project maintainers feel that the .NET Foundation has lied to them and abused their administrative privileges. I understand their frustration. 

I would like to ask everyone to take a deep breath, not to assume the worst, hear out both sides, and keep an open mind.

Please give the board time. Remember, we just had an election (that hopefully all .NET Foundation members participated in). It takes time for the new board members to onboard to the foundation as the old board members leave. Also remember, that most of the board are not paid to be a member of the board. They're volunteering their time and effort.

Also, at least to my knowledge, please realize that the board discussing these things internally before talking to the community is due to the way the bylaws and the .NET Foundation run. There needs to be a quorum of members to vote on decisions surrounding the .NET Foundation. As .NET Foundation members, we voted for these board members to represent us. Please don't assume the worst. 

Finally, I'd like to say one thing I think the .NET Foundation and the community needs to continue to improve is communication. I've been frustrated in the past with the .NET Foundation and communication. As a community member, I did nothing to try to rectify and express my concerns until recently. There are growing pains. The foundation is still relatively young. The first openly elected board was just elected in March of 2019 after the announcement of the open membership of the foundation in December 2018. So, hopefully we can use this as a learning experience to have a more open dialog between the members and the board.

**Reply by dustinmoris** [reply]
> Please give the board time. Remember, we just had an election (that hopefully all .NET Foundation members participated in). It takes time for the new board members to onboard to the foundation as the old board members leave.

That is such a poor excuse. It wasn't the entire board that got replaced. Also there is nothing holding old board members back from chiming in or providing reassurance on the whys of their actions within the DNF.

The fact that there is such much radio silence from the DNF on both public GitHub discussions just shows how out of touch they are with their members. If this was a member run foundation then they would be at the centre of this discussion, but nobody except some corporate Microsoft employees at the DNF have a clue, because they are the ones who really call the shots and they prefer to say as little as possible because their number one priority is to protect their employer and not stand up for members of the DNF.

It's complete shambles.

**Reply by JimBobSquarePants** [reply]
The Foundation has been around since 2014. There’s no excuse
